state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1051,6 +1054,12 @@ impl<T: Config> SwappingApi for Pallet<T> {
 			Ok(output_amount)
 		})
 	}
+
+	fn add_flip_to_burn(amount: AssetAmount) {
+		FlipToBurn::<T>::mutate(|total| {
+			total.saturating_accrue(amount);
+		});
+	}
```

Comment:
It might be easier/cleaner to just move the FLipToBurn storage into the Swapping pallet rather than extend the SwappingApi with this arbitrary `add_flip_to_burn` method.